### DISCUSSION TITLE
EGSnrc 2023 has been released

### DISCUSSION THREAD

**Discussion by ftessier** [nrc-cnrc/EGSnrc#953] [Announcements]
**The 2023 version of EGSnrc** has been released on February 15 (there was no release in 2022). Notable changes from the previous version are listed in the [release notes](https://github.com/nrc-cnrc/EGSnrc/wiki/Release-notes). Please review the recommended way to [cite EGSnrc](https://github.com/nrc-cnrc/EGSnrc/wiki/Citing-EGSnrc). To install EGSnrc, clone the repository and follow the [installation instructions](https://github.com/nrc-cnrc/EGSnrc/wiki/Installation-overview) for your operating system.

**Comment by karl63** [comment]
This version contains a code of conduct. What do we need it for? We have been  living for many years without it. Why now? uncorect behavior if we want to call it that is always "self correcting". If someone writes "I pretend the solution for my problem and I want it tomorrow  " or if someone posts a question  and one sees that she did not try at least to solve the problem no one will answer . It is as simple as that. Also I never saw  when some wrote "I did this and this and this, but then I got That and that and that error message" that the answer was "You are a complete idiot if you are not even able to solve this simple problem by yourself" Or the like.  
On the covenant website I read "We recognize that the adoption of a code of conduct can be a source of confusion or even a n g e r for some people in an open source community" That is exactly what I feel. The community is considered  as a bunch of badly behaved children on which the woke agenda must be enforced.  Up to one year ago I was a member of the AAPM. When they required me to sign a code of conduct, I left the association.

**Reply by ftessier** [reply]
Indeed we added a code of conduct. It clarifies what you can expect when participating in the EGSnrc project. We are certainly not asking anyone to sign anything.

**Comment by AndrewFielding** [comment]
Installed the 2023 version for Windows. When I try and build and compile accelerator models (including supplied examples) in BEAMnrc, builds ok, but compilation fails. The same accelerators build and compile in 2021 version and older versions ok. The installation instructions still say you should use minGW compilers v7.3.0, is this still the case for 2023 version?

**Reply by ftessier** [reply]
@rtownson is this related to #972?

**Reply by rtownson** [reply]
> @rtownson is this related to #972?

Possibly, yes. 

Could you share your installation log, found in HEN_HOUSE/log? The newest minGW compilers do not work, I'm not sure where exactly the cutoff is. So depending on what you see in the log, you could try an older version.

**Reply by AndrewFielding** [reply]
I'm using minGW 7.3.0, which is the recommended version. EGSnrc 2021 and earlier all compile ok with this.

I think its the COMPILE_TIME error ftessier referred to in #972, see the screenshot below

![image](https://github.com/nrc-cnrc/EGSnrc/assets/3496901/c3315c9a-3d68-43f0-b0d4-273e662ff6d8)

**Reply by rtownson** [reply]
If you are comfortable, you can try making this change yourself: [HEN_HOUSE/specs/all_common.spec](https://github.com/nrc-cnrc/EGSnrc/pull/972/files#diff-761d22ed0e286d1953f225081a37c7d63bb29d476ae73dfb1f2ecd460f4829b7)

Just change line 145, removing `$(shell cmd /C tzutil /g)`. Then recompile your accelerator.

**Reply by AndrewFielding** [reply]
I made that change and had the same problem. Found that the same command is in the HEN_HOUSE/specs/beamnrc.spec file as well, change needs to be in that file also. Then acccelerator seems to compile ok.

Thanks for your help.

**Reply by rtownson** [reply]
Thanks for pointing this out, I've added that change to the fix in the PR.

**Comment by Laymandamn** [comment]
作者你好，我也是egs新用户使用者，正在通过学习手册进行学习，我使用的是windows系统进行学习的，当我进行到egs_view slab.egsinp slab.ptracks这一步时，我发现我显示出来的图像是一个平面和一个三维坐标系，那些红色的粒子轨迹并没有显示出来，这使我非常苦恼。当我用git bash 对egs_view slab.ptracks进行操作时，会显示segmentation fault，并出现闪退现象。还有另一个问题就是当我运行到 mingw32-make这一步时出现了一个错误/usr/bin/sh: -c: line 2: syntax error: unexpected end of file，这是不是对上一个问题出现的罪魁祸首
![1](https://github.com/user-attachments/assets/0e0eeb19-1592-40ce-823d-14963eb075d2)
![2](https://github.com/user-attachments/assets/8c23b9d3-dd8b-40c4-aaa5-8cc9c4e60550)
![3](https://github.com/user-attachments/assets/5e9a18c2-1979-49ec-b883-e91a7dc55bc1)

**Reply by rtownson** [reply]
Hello, please create a new discussion by clicking on the green "New Discussion" button on the [Discussions](https://github.com/nrc-cnrc/EGSnrc/discussions) page. When you post there, include the full output from running the egs_app simulation - it looks like it is running but not creating the ptracks file - does the file exist? The command `egs_view slab.ptracks` will never work, your first command with both the .egsinp and .ptracks is correct.
